src/nl_robot_console/console.py

```diff
@@ -265,6 +264,9 @@ def main():
         rospy.init_node("nl_robot_console")
         repl = REPL(args.grammar, debug=args.debug)
 
+        # when user hits ctrl+d in console, cancel all acctions
```

Comment:
I don't think this is the correct place.

Reply:
You should use `rospy.on_shutdown(h)` so you can actually still send messages.

Reply:
I think it is, but you can use rospy.on_shutdown with a callback. 

But I can imagine that the REPL object catches ctr-D and other signals.

Reply:
This is already going into way too much detail, the question is still "is this the behavior we are looking for?"

Reply:
ctrl-D for exit 9as with many other REPLs): YES!

Reply:
When a goal is running, the user does not have input control, but must wait for the action's result. During that state, the running goal should be canceled at a KeyboardInterrupt. cmd.Cmd already handles EOF (Ctrl+D), but this does not have to cancel any actions, because the user can only exit the program after any running action returned a result (either canceled, succeeded or failed).